### PR TITLE
docs: rewrite BUILD-manual and drop configure-manual support

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

File: docs/BUILD-manual.md [hunk 1]
```diff
@@ -20,73 +20,42 @@ git clone https://github.com/nemtech/catapult-client.git
...
-> If you only want to download the dependencies (without building and installing them) enter the command ``scripts/configure-manual.sh download deps`` instead. Type ``scripts/configure-manual.sh --help`` for the summary of all commands.
-
-## Step 3: Prepare build directory
-
-To create or update the ``_build`` directory type into a terminal:
+> If you want to build with clang, add the `--use-clang` flag.
 
 ```sh
-scripts/configure-manual.sh
+PYTHONPATH="./scripts/build" python "./scripts/build/installDepsLocal.py" \
```
Comment (segfaultxavi) [inline docs/BUILD-manual.md:31]: I've got `python3` but not `python` and this line is failing. Using `python3` works.

File: docs/BUILD-manual.md [hunk 2]
```diff
@@ -20,73 +20,42 @@ git clone https://github.com/nemtech/catapult-client.git
...
+	--target "./_deps" \
+	--versions "./scripts/build/versions.properties" \
+	--download \
+	--build
 ```
```
Comment (segfaultxavi) [inline docs/BUILD-manual.md:36]: This line fails for me (Ubuntu):
```
[*] <mkdirs> _deps/source True
[*] <chdir> _deps/source
[x] building all dependencies
[*] <chdir> _deps/source/boost
Traceback (most recent call last):
  File "./scripts/build/installDepsLocal.py", line 136, in <module>
    main()
  File "./scripts/build/installDepsLocal.py", line 129, in main
    builder.build_boost()
  File "./scripts/build/installDepsLocal.py", line 45, in build_boost
    self.environment_manager.chdir(self.target_directory / SOURCE_DIR_NAME / 'boost')
  File "/home/xartigas/repos/catapult-client/scripts/build/environment.py", line 57, in chdir
    os.chdir(path)
FileNotFoundError: [Errno 2] No such file or directory: '_deps/source/boost'
```
Looks like it tries to cd into `_deps/source/boost` when it's already in `_deps/source`.

**Review by segfaultxavi — CHANGES_REQUESTED** [review]
- I would use
  ```
  git clone git@github.com:symbol/catapult-client.git` 
  ```
  To avoid requiring GitHub credentials.

- Prerequisites is missing Python (and its version).

I couldn't finish the review due to the build failure.

**Review by gimer — APPROVED** [review]

File: docs/BUILD-manual.md (hunk 2 above)
Comment (Jaguar0625) [reply docs/BUILD-manual.md:36]: try now

File: docs/BUILD-conan.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ as this will probably take *a bit*.
 ```sh
 conan remote add nemtech https://catapult.jfrog.io/artifactory/api/conan/ngl-conan
 
-git clone https://github.com/nemtech/catapult-client.git
+git clone git@github.com:symbol/catapult-client.git
```
Comment (gimer) [inline docs/BUILD-conan.md:56]: I don't remember if this will work if you do not have access...

File: docs/BUILD-manual.md (hunk 2 above)
Comment (segfaultxavi) [reply docs/BUILD-manual.md:36]: Fixed.

**Comment by segfaultxavi** [comment]
Python3 is still missing from the prerequisites, right?

File: docs/BUILD-conan.md (hunk 3 above)
Comment (Jaguar0625) [reply docs/BUILD-conan.md:56]: @segfaultxavi ?

Comment (segfaultxavi) [reply docs/BUILD-conan.md:56]: I'm no GH expert but using the https URL asks for my credentials, whereas the git@ works without them.
What do you mean "if you do not have access"?

Comment (segfaultxavi) [reply docs/BUILD-conan.md:56]: I definitely used `nemtech` in one of my previous experiments. Now both methods work for me, even after removing `~/.ssh`.

Sooo... ¯\\\_(ツ)\_/¯

File: docs/BUILD-manual.md [hunk 4]
```diff
@@ -16,77 +17,46 @@ These instructions have been verified to work on Ubuntu 20.04 with 8 GB of RAM a
...
+To create or update the ``_build`` directory type into a terminal:
 
-Regardless of whether the tools are installed globally or not, their dependencies must be accessible before running them so make sure to update ``LD_LIBRARY_PATH``:
+> **NOTE**:
+> When building with clang, add `-DENABLE_FUZZ_BUILD=ON` to build with UBSAN and ASAN (recommended for development).
 
 ```sh
-export LD_LIBRARY_PATH=$CAT_DEPS_DIR/boost/lib:$CAT_DEPS_DIR/facebook/lib:$CAT_DEPS_DIR/google/lib:$CAT_DEPS_DIR/mongodb/lib:$CAT_DEPS_DIR/zeromq/lib
+mkdir -p _build && cd _build
+cmake BOOST_ROOT="../_deps/boost" cmake .. \
```
Comment (segfaultxavi) [inline docs/BUILD-manual.md:48]: Why are there two `cmake` in this line?

**Review by segfaultxavi — CHANGES_REQUESTED** [review]
Fails for me during `cmake`:
```
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY) (Required is exact version "1.10.0")
```

After "fixing" them by adding `-DGTEST_INCLUDE_DIR="../_deps/google/include/gtest/" -DGTEST_LIBRARY="../_deps/google/lib/" -DGTEST_MAIN_LIBRARY="../_deps/google/lib/"` it then fails with:
```
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR) (Required is exact version "1.1.1")
```

File: docs/BUILD-conan.md (hunk 3 above)
Comment (Jaguar0625) [reply docs/BUILD-conan.md:56]: ok, will revert

File: docs/BUILD-manual.md (hunk 4 above)
Comment (Jaguar0625) [reply docs/BUILD-manual.md:48]: first one is wrong

**Review by segfaultxavi — APPROVED** [review]

**Pull request merged**